### PR TITLE
Add config option to announce tournaments

### PR DESCRIPTION
**Pull request by sirDonovan**
When tournaments are created in the specified rooms, they will be
announced in the server's main tournament room

http://i.imgur.com/2DuTRUt.png

### CODE REVIEW

**Comment by TheImmortal** [comment]
:+1: Slight nitpick from me is that I think only the room name should be linked and not the word "room". Also, a period at the end.

**Comment by TheImmortal** [comment]
@Zarel is this fine?

File: config/config-example.js [hunk 1]
```diff
@@ -160,6 +160,15 @@ exports.customavatars = {
 	//'userid': 'customavatar.png'
 };
 
+// Tournament announcements
+// When tournaments are created in rooms listed below, they will be announced in
+// the server's main tournament room (either the specified tourroom or by default
+// the room 'tournaments')
+exports.tourroom = '';
+exports.tourannouncements = {
+	//'roomid': 1
```
Comment (Zarel) [inline config/config-example.js]: This should be an array. The `:1` micro-optimization stuff shouldn't be exposed to config.

File: tournaments/index.js [hunk 2]
```diff
@@ -879,7 +879,13 @@ CommandParser.commands.tournament = function (paramString, room, user) {
 		}
 
 		var tour = createTournament(room, params.shift(), params.shift(), params.shift(), Config.istournamentsrated, params, this);
-		if (tour) this.privateModCommand("(" + user.name + " created a tournament in " + tour.format + " format.)");
+		if (tour) {
+			this.privateModCommand("(" + user.name + " created a tournament in " + tour.format + " format.)");
+			if (Config.tourannouncements && Config.tourannouncements[room.id]) {
+				var tourRoom = Rooms.search(Config.tourroom || 'tournaments');
+				if (tourRoom) tourRoom.addRaw('<div class="infobox">' + Tools.getFormat(tour.format).name + ' tournament created in the <u><a href="/' + room.id + '" class="ilink">' + room.title + '</a></u> room.</div>');
```
Comment (Zarel) [inline tournaments/index.js]: I'd say "A [FORMAT] tournament was created in [ROOM]."

Comment (Zarel) [inline tournaments/index.js]: Generally, for UI messages, more concise is better. Users will be reading it hundreds of times, after all.

Comment (Zarel) [inline tournaments/index.js]: `<u>`? Just remove class `ilink`...

File: config/config-example.js (hunk 1 above)
Comment (xfix) [inline config/config-example.js]: I miss that JavaScript is not let's say, Python. Allowing you to write something like `{'tournaments'}` would be quite neat.

File: tournaments/index.js (hunk 2 above)
Comment (TheImmortal) [inline tournaments/index.js]: No "A" in front please. I'd say: "[FORMAT] tournament created in [ROOM]." We can drop the "was".

**Comment by TheImmortal** [comment]
>I'd say "A [FORMAT] tournament was created in [ROOM]."

No "A" in front please. I'd say: "[FORMAT] tournament created in [ROOM]." We can drop the "was".

**Comment by Zarel** [comment]
Works for me, TI.

**Comment by Zarel** [comment]
New idea: bold format and room, and just linkify the entire line (but make it an `ilink`).

**Comment by TheImmortal** [comment]
I'd like to see a screenshot of that before I can comment.

**Comment by sirDonovan** [comment]
Screenshot: http://imgur.com/pQVMf88

**Comment by TheImmortal** [comment]
That's fine with me. Don't forget the period when you change it here!

**Comment by sirDonovan** [comment]
Oh right, thanks. Updated now

**Comment by Zarel** [comment]
>Omitting the period from the link

...if you're going to link the entire line, that's just _weird_...

**Comment by sirDonovan** [comment]
From what I've been told and seen around the web, punctuation at the end of a sentence shouldn't be included in a link. I can change it if you want it that way here.

**Comment by TheImmortal** [comment]
Agree with Zarel, it is weird. And we already set a precedent by including it in battle started messages.

**Comment by sirDonovan** [comment]
Alright, updated.

**Comment by Zarel** [comment]
> punctuation at the end of a sentence shouldn't be included in a link.

Well, that's only if the link is inline. This is pseudo-block-level.

**Comment by sirDonovan** [comment]
Fair enough. Any other changes that should be made?

**Comment by TheImmortal** [comment]
Merging this. If Zarel or someone else has an issue later on, we can change it then.

**Pull request merged**